public/app/core/components/Select/DashboardPicker.tsx

```diff
@@ -56,19 +57,39 @@ export const DashboardPicker = ({
     (async () => {
       // value was manually changed from outside or we are rendering for the first time.
       // We need to fetch dashboard information.
-      const res = await getDashboardAPI().getDashboardDTO(value, undefined);
-      const formatted = ResponseTransformers.ensureV1Response(res);
+      // when using getDashboardAPI, if isV2Mode is not set, we will always return the v1 schema version
+      // isV2Mode is set to true when the feature toggle of dashboardSchemaV2 and useV2DashboardsAPI is enabled
+      // We could create a different feature toggle with better naming if we want, but the idea of using
+      // dashboardSchemaV2 is to indicate that this component is ready to process the new schema
+      const isUIReadyForV2 = config.featureToggles.useV2DashboardsAPI;
+
+      if (isUIReadyForV2) {
+        const resWithSchemaV2 = await getDashboardAPI('v2').getDashboardDTO(value, undefined);
```

Comment:
this seems messy to have check feature toggle and pass in different params and have conditional logic everywhere we use getDashboardAPI 

Hm.. getDashboardAPI should just return different implementations of the same interface. 

Ideally we should return a type that both the v1 and v2 apis can be translated to (except maybe the spec type will be different). but all the meta field we should easily be able to translate to the same type no matter if you use v1 or v2

Reply:
> Ideally we should return a type that both the v1 and v2 apis can be translated to (except maybe the spec type will be different). but all the meta field we should easily be able to translate to the same type no matter if you use v1 or v2

that would be an ultimate goal longer term, but... see this PR that explores this https://github.com/grafana/grafana/pull/97559